src/exercises/cheriabi/README.md

```diff
@@ -0,0 +1,195 @@
+# CheriABI Showcase
+
+This exercise demonstrates several aspects of CheriABI, which defines how
+CheriBSD processes are constructed, how function calls are made, how a process
+interacts with the kernel through system calls, and other such foundational
+details.
+
+## The Kernel Voluntarily Honors Capability Bounds
+
+`kern-read-over.c` demonstrates a (potential) loss of spatial safety when
+pointers are passed from userspace to the kernel.  The kernel, by convention,
+has full access to all of userspace memory.  Even when CheriBSD is running
+CheriABI programs, this is true: the kernel holds a capability with full RWX
+access to all userspace addresses.  Therefore, the kernel can act as a
+*confused deputy*, accessing memory with its legitimate authority but without
+intent.
+
+1. Compile `kern-read-over.c` for both the baseline (`kern-read-over-baseline`)
+   and CHERI-enabled (`kern-read-over-cheri`) architectures.
+
+2. Run these programs and observe their outputs.
+
+3. Focusing on the `read()` system call, what happens in the two versions of the
+   program.  When, in particular, does it look like the CHERI version notices
+   something is amiss?
+
+4. If you have done the [inter-stack-object buffer overflow
+   exercise](../buffer-overflow-stack), contrast the behaviors of the two
+   CHERI-enabled programs.
+
+## The Process Memory Map
```

Comment:
I wonder if the more basic point that memory mappings are returned via bounded pointers should be mentioned.